### PR TITLE
[ENG-282] Add Crypto docs

### PR DESCRIPTION
**Pull request by brxken128**
This PR adds detailed (but not fully extensive) documentation regarding the key manager and encryption in general.

It is a little out of sync with the key manager found in #450, but once we refine the key manager they will line up perfectly.

I have proof read this, but another pair of eyes never hurts.

### CODE REVIEW

**Comment by maxichrome** [comment]
Gonna make a proofreading/editing run through this before merge. Have only read bits thus far but doesn't look like it'll be that big a job

**Comment by brxken128** [comment]
> Gonna make a proofreading/editing run through this before merge. Have only read bits thus far but doesn't look like it'll be that big a job

Thank you! Grammar is quite evidently not my forte.

**Review by oscartbeaumont — APPROVED** [review]

File: docs/developers/architecture/key-manager.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Support for changing a master password will be added at a later date, but for now it is not a priority.
+
+## Performance
+
+Designing the key manager, and our key system as a whole, was a difficult task. It needs to be three things: secure, performant, and not annoying for the user. The main way to do this is to implement a hierarchical system, similar to what we have done.
+
+We chose specific password hashing parameters in order to provide a great performance/time relationship, and even "paranoid" keys don't take too long to hash (even though they use lots of memory).
+
+## Internal Architecture
```
[the remark this answers is not in the document]
Comment (brxken128) [reply docs/developers/architecture/key-manager.md:35]: That's very true - a lot of the content I wrote is either redundant, or just not suitable for documentation. I'll bring it up with @maxichrome when we next try and get these docs sorted (this PR should've probably been a draft, my apologies).

**Comment by brxken128** [comment]
Closing this in favour of our new (and improved) crypto docs.

**Pull request closed without merging**